src/libraries/System.Private.CoreLib/src/System/Diagnostics/CodeAnalysis/DynamicallyAccessedMemberTypes.cs

```diff
@@ -0,0 +1,96 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Diagnostics.CodeAnalysis
+{
+    /// <summary>
+    /// Specifies the types of members that are dynamically accessed.
+    ///
+    /// This enumeration has a <see cref="FlagsAttribute"/> attribute that allows a
+    /// bitwise combination of its member values.
+    /// </summary>
+    [Flags]
+    public enum DynamicallyAccessedMemberTypes
+    {
+        /// <summary>
+        /// Specifies no members.
+        /// </summary>
+        None = 0,
+
+        /// <summary>
+        /// Specifies the default, parameterless public constructor.
+        /// </summary>
+        DefaultConstructor = 0x0001,
+
+        /// <summary>
+        /// Specifies all public constructors.
+        /// </summary>
+        PublicConstructors = 0x0002 | DefaultConstructor,
+
+        /// <summary>
+        /// Specifies all non-public constructors.
+        /// </summary>
+        NonPublicConstructors = 0x0004,
+
+        /// <summary>
+        /// Specifies all public methods.
+        /// </summary>
+        PublicMethods = 0x0008,
+
+        /// <summary>
+        /// Specifies all non-public methods.
+        /// </summary>
+        NonPublicMethods = 0x0010,
+
+        /// <summary>
+        /// Specifies all public fields.
+        /// </summary>
+        PublicFields = 0x0020,
+
+        /// <summary>
+        /// Specifies all non-public fields.
+        /// </summary>
+        NonPublicFields = 0x0040,
+
+        /// <summary>
+        /// Specifies all public nested types.
+        /// </summary>
+        PublicNestedTypes = 0x0080,
+
+        /// <summary>
+        /// Specifies all non-public nested types.
+        /// </summary>
+        NonPublicNestedTypes = 0x0100,
+
+        /// <summary>
+        /// Specifies all public properties.
+        /// </summary>
+        PublicProperties = 0x0200,
+
+        /// <summary>
+        /// Specifies all non-public properties.
+        /// </summary>
+        NonPublicProperties = 0x0400,
+
+        /// <summary>
+        /// Specifies all public events.
+        /// </summary>
+        PublicEvents = 0x0800,
+
+        /// <summary>
+        /// Specifies all non-public events.
+        /// </summary>
+        NonPublicEvents = 0x1000,
+
+        /// <summary>
+        /// Specifies all members.
+        /// </summary>
+        All = DefaultConstructor | PublicConstructors | NonPublicConstructors |
```

Comment:
Per the discussion in the meeting - I thought the agreement was to make All a special value and treat it separately (so I would expect 0xFFFFFFFF) - the code would specifically first check for that value and include everything. As oppose to rely on the fact that bitwise checks would "work". To avoid the potential problem with adding more bits in the future.

Reply:
Good point. I missed this in the meeting. I rewatched it and agree that was the conclusion.

My thinking was that defining `All` the way I am above is better for backwards compatibility, in general.

If we add another bit in the future (say `Attributes`), if an app/library is compiled against `net5.0` and used the `All` value, they would get `8191` as the value compiled into their code. When their code is executed against something that only understands `net5.0` values it behaves the same as when they execute against something that understands `net6.0` with the new bits. They don't suddenly change behavior just because a new value was added to the enum.

Only when the library/application author recompiles their code against the updated enum, do they get the new behavior.

This has the advantage that when a piece of code is not under development (i.e. the same compiled assembly used in multiple places), it behaves the same. And only when the code is under development (i.e. being re-compiled on an updated version) do we actually change behavior, and the developer can test the new behavior works in the new version.

However, in this specific instance we are saying `All` means everything in the `Type` now and in the future. Even if we add new values to this enum in the future, `All` means those new values now as well. (Basically, the linker won't touch the type at all, and can never warn about Reflection on that type.)